File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer=0;
    
    public static void dfs(int node,int[]numbers,int target,int sum){
       
        if(node==numbers.length){
            if (sum==target){
                answer++;
            }
            return;
        }
        dfs(node+1,numbers,target,sum+numbers[node]);
        dfs(node+1,numbers,target,sum-numbers[node]);
    }
    
    
    public int solution(int[] numbers, int target) {
        dfs(0,numbers,target,0);
        return answer;
    
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer =0 

def dfs(numbers,target,index,curr):
    
    global answer
    
    if index == len(numbers) and curr== target:
        answer+=1
        return
    
    elif index == len(numbers):
        return
    
    dfs(numbers,target,index+1,curr-numbers[index])
    dfs(numbers,target,index+1,curr+numbers[index])


def solution(numbers, target):
    
    global answer 
    
    dfs(numbers,target,0,0)

    return answer


